tests/queries/test_q.py

```diff
@@ -320,10 +322,18 @@ def test_rawsql(self):
         replaced by its value. In this case, Q.check() logs a warning and
         return True.
         """
+
+        # It's important that we use the same connection for q.check() and is_usable()
+        connection_alias = DEFAULT_DB_ALIAS
+
         q = Q(RawSQL("price > %s", params=(20,), output_field=BooleanField()))
         with self.assertLogs("django.db.models", "WARNING") as cm:
-            self.assertIs(q.check({"price": 10}), True)
+            self.assertIs(q.check({"price": 10}, using=connection_alias), True)
         self.assertIn(
             f"Got a database error calling check() on {q!r}: ",
             cm.records[0].getMessage(),
         )
+
+        # See ticket #35712
+        # We must leave connection in a usable state
+        self.assertTrue(connections[connection_alias].is_usable())
```

Comment:
I guess we could have used `django.db.connection` as well as that what `Q.check` will default to if not specified.

```python
self.assertTrue(connection.is_usable())
```

Reply:
Yes, that would have been more concise. I just thought it was a good idea to "mark" the `q.check()` call, so that someone wanting to change that line (ie. to use another connection) would think twice. But perhaps I was over-thinking. No one should be changing the existing test code, anyway.